deepfence_worker/ingesters/secrets.go

```diff
@@ -68,7 +73,24 @@ func secretsToMaps(data []ingestersUtil.Secret) []map[string]map[string]interfac
 		for k, v := range utils.ToMap(i.Match) {
 			secret[k] = v
 		}
-		secret["node_id"] = utils.ScanIdReplacer.Replace(fmt.Sprintf("%v:%v", i.Rule.ID, i.Match.FullFilename))
+
+		entityId := ""
+		if _, ok := secret["scan_id"]; ok {
+			scanId := secret["scan_id"].(string)
+			var err error
+			entityId, err = getEntityIdFromScanID(scanId, string(utils.NEO4J_SECRET_SCAN), tx)
+			if err != nil {
+				//TODO::Should we treat this as hard error and return?
```

Comment:
yes we should so the ingester can retry and no data is being lost

Reply:
Done.